File: 7th/jongmlee/Sticker9465.java
import java.util.*;

public class Sticker9465 {
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);

		int t = sc.nextInt();
		while (t-- > 0) {
			int n = sc.nextInt();
			int[][] sticker = new int[2][n + 1];
			for (int i = 0; i < 2; i++) {
				for (int j = 1; j <= n; j++) {
					sticker[i][j] = sc.nextInt();
				}
			}
			int[][] dp = new int[2][n + 1];
			dp[0][1] = sticker[0][1];
			dp[1][1] = sticker[1][1];
			if (n == 1) {
				System.out.println(Math.max(dp[0][1], dp[1][1]));
				continue;
			}
			dp[0][2] = dp[1][1] + sticker[0][2];
			dp[1][2] = dp[0][1] + sticker[1][2];
			for (int i = 3; i <= n; i++) {
				dp[0][i] = Math.max(dp[1][i - 1], Math.max(dp[0][i - 2], dp[1][i - 2])) + sticker[0][i];
				dp[1][i] = Math.max(dp[0][i - 1], Math.max(dp[1][i - 2], dp[0][ i - 2])) + sticker[1][i];
			}
			System.out.println(Math.max(dp[0][n], dp[1][n]));
		}
	}
}

File: 7th/jimchoi/sticker_9465.cpp

#include <iostream>
using namespace std;

int t;
int n;
int d[3][100001];
int sticker[3][100001];

int main()
{
    cin >> t;
    for(int i=0; i<t; i++)
    {
        cin >> n;
        for(int i=0; i<n; i++)
            cin >> sticker[1][i+1];
        for(int i=0; i<n; i++)
            cin >> sticker[2][i+1];
        d[1][1] = sticker[1][1];
        d[2][1] = sticker[2][1];
        d[1][2] = sticker[1][2] + sticker[2][1];
        d[2][2] = sticker[2][2] + sticker[1][1];
        for(int i=3; i<=n; i++)
        {
            d[1][i] = max(max(d[1][i-2], d[2][i-1]), d[2][i-2]) + sticker[1][i];
            d[2][i] = max(max(d[2][i-2], d[1][i-1]), d[1][i-2]) + sticker[2][i];
        }
        cout << max(d[1][n], d[2][n]) << endl;

    }
}
